cfpb/capital-framework
# Updated form inputs to atomic design principles

Updates form inputs as well as a few necessary parent molecules and organisms of the inputs.

## Additions

- Added `text-input` atom to combine all text style inputs as one and reduce specificity of existing inputs.
- Added `checkbox` and `radio` atoms to simplify styles. Removes the need for hiding the input and relying on pseudo elems on the label.
- Added `label` atom with heading modifier.

## Removals

- Removed form variables from cf-core

## Changes

- Updated form variable naming scheme to be more consistent with each other and our BEM style principles.

## Testing

- run `npm run cf-link`
- in a tab with the sandbox repo run the following:
    - checkout the `form-examples` [branch](https://github.com/cfpb/capital-framework-sandbox/pull/23) 
    - run `npm run cf-link`
    - run `npm run build`
    - run `npm start`
    - navigate to http://localhost:3000/components/cf-forms/ and check out the text inputs at the top and the Atomic Filter Form at the bottom. Ignore all the broken bits, those are fixed and ready for a follow up PR.

## Review

- @mistergone 
- @cfarm 
- @KimberlyMunoz 
- @Scotchester 

## Screenshots

__Labels__

![screen shot 2016-08-30 at 4 01 10 pm](https://cloud.githubusercontent.com/assets/1280430/18106858/fe600d04-6eca-11e6-941a-fcdb307f95e2.png)

__Basic inputs__

![screen shot 2016-08-30 at 4 01 30 pm](https://cloud.githubusercontent.com/assets/1280430/18106867/0d98dcc4-6ecb-11e6-85a5-4170864e398a.png)

__Input states__

![screen shot 2016-08-30 at 4 01 52 pm](https://cloud.githubusercontent.com/assets/1280430/18106880/173e7c7a-6ecb-11e6-9b9b-8e0d38902168.png)

__Filter form example__

![screen shot 2016-08-30 at 4 02 22 pm](https://cloud.githubusercontent.com/assets/1280430/18106895/29d2f37a-6ecb-11e6-8333-9a2231563855.png)

## Notes

- ⚠️  __WARNING:__ The sandbox branch breaks a lot of the other form elements. There are upcoming PRs to fix those, please ignore for the time being.

## Checklist
